src/tiled/mainwindow.cpp

```diff
@@ -608,7 +609,37 @@ bool MainWindow::openFile(const QString &fileName, FileFormat *fileFormat)
     Document *document = nullptr;
 
     if (MapFormat *mapFormat = qobject_cast<MapFormat*>(fileFormat)) {
-        document = MapDocument::load(fileName, mapFormat, &error);
+        MapDocument *mapDocument =  MapDocument::load(fileName, mapFormat, &error);
+        document = mapDocument;
+
+        bool hasNonEmbeddedTemplateGroups;
```

Comment:
Variable is missing an initialization. Not that it matters, because the variable isn't really used. Remove?